libraries/commerce/scanner/actions/handleQrCode.js

```diff
@@ -0,0 +1,93 @@
+import Scanner from '@shopgate/pwa-core/classes/Scanner';
+import { historyPush } from '@shopgate/pwa-common/actions/router';
+import { fetchPageConfig } from '@shopgate/pwa-common/actions/page';
+import { getPageConfigById } from '@shopgate/pwa-common/selectors/page';
+import showModal from '@shopgate/pwa-common/actions/modal/showModal';
+import { fetchProductsById, getProductById } from '@shopgate/pwa-common-commerce/product';
+import { fetchCategory, getCategoryById } from '@shopgate/pwa-common-commerce/category';
+import {
+  QR_CODE_TYPE_CATEGORY,
+  QR_CODE_TYPE_COUPON,
+  QR_CODE_TYPE_HOMEPAGE,
+  QR_CODE_TYPE_PAGE,
+  QR_CODE_TYPE_PRODUCT,
+  QR_CODE_TYPE_PRODUCT_WITH_COUPON,
+  QR_CODE_TYPE_SEARCH,
+} from '../constants';
+import { parse2dsQrCode } from '../helpers';
+
+/**
+ * Handle qr code
+ * @param {string} [payload=''] qr code payload
+ * @return {Function} A redux thunk.
+ */
+export default payload => async (dispatch, getState) => {
```

Comment:
At the moment you push a new route when the qr code contains a valid payload. As a result the user always has the scanner inside of the browser history.
I've discussed this with our PO, and in her opinion it might be better to `replace` the scanner route with the new route.
There is one exception - at `QR_CODE_TYPE_COUPON` you we need to pop the route, so that the user is at the previous page again when he sees the feedback modal.

Please do the same adjustment for the barcode handler.